src/transactions/test/PathPaymentStrictSendTests.cpp

```diff
@@ -511,18 +511,29 @@ TEST_CASE("pathpayment strict send", "[tx][pathpayment]")
 
     SECTION("issuer missing")
     {
+        // look at the SECTION "uses all offers in a loop" for a
+        // successful path payment test with no issuers
+
         auto market = TestMarket{*app};
         auto source = root.create("source", minBalance1);
         auto destination = root.create("destination", minBalance1);
         source.changeTrust(idr, 20);
         destination.changeTrust(usd, 20);
         gateway.pay(source, idr, 10);
 
-        SECTION("send issuer missing")
-        {
-            auto sendIssuerMissing = [&](bool issuerCheckExists) {
+        for_versions_from(12, *app, [&] {
+            uint32_t ledgerVersion;
+            {
+                LedgerTxn ltx(app->getLedgerTxnRoot());
+                ledgerVersion = ltx.loadHeader().current().ledgerVersion;
+            }
+
+            auto isPreV13 = [ledgerVersion]() { return ledgerVersion < 13; };
```

Comment:
Why use a lambda here? This code is still extremely repetitive and could benefit from refactoring. A more useful lambda would be something that encapsulates, for example, lines 536-549 and then reuses that for 555-568 and 574-587.

Reply:
Good point. Fixed.